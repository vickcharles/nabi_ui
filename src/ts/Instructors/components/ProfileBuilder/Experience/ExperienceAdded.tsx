import * as React from 'react';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import Typography from 'material-ui/Typography';
import { ExperienceType }  from '../../../model';
import Divider from 'material-ui/Divider';

/**
 * Props for ExperienceAdded
 * @interface ExperienceAddedProps
 */
interface ExperienceAddedProps extends ExperienceType {
  deleteExperience: (experienceId: any) => void;
}

/**
 * ExperienceAdded
 * @extends React.StatelessComponent<ExperienceAddedProps>
 */
const ExperienceAdded: React.StatelessComponent<ExperienceAddedProps> = props => {
  const fromMonth = props.fromMonth.charAt(0).toUpperCase() + props.fromMonth.slice(1);
  const toMonth = props.toMonth.charAt(0).toUpperCase() + props.toMonth.slice(1);
  return (
    <div>
      <div className="selected-education">
        <IconButton className="nabi-float-right" aria-label="Delete" onClick={() => props.deleteExperience(props.id)}>
          <DeleteIcon />
        </IconButton>
        <Typography className="nabi-margin-top-small" variant="body2">{props.jobTitle.toUpperCase()}</Typography>
        <Typography className="">{props.employer}</Typography>
        <Typography className="">{`${fromMonth}, ${props.fromYear} - ${toMonth}, ${props.toYear}`}</Typography>
        <Typography className="">
          {props.location}
        </Typography>
      </div>
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default ExperienceAdded;
