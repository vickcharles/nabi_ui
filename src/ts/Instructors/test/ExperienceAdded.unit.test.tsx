import * as React from 'react';
import ExperienceAdded from '../components/ProfileBuilder/Experience/ExperienceAdded';
import { shallow } from 'enzyme';

describe('ExperienceAdded', () => {
  let wrapper: any;
  const deleteExperience: () => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <ExperienceAdded
          employer="Oracle"
          jobTitle="Software Engineer"
          fromMonth="sep"
          fromYear="2017"
          toMonth="sep"
          toYear="2018"
          location="Cambridge, MA"
          deleteExperience={deleteExperience}
        />
      );
    });
    
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
