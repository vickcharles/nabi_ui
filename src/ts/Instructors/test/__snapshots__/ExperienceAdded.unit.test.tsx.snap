// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceAdded Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExperienceAdded
    deleteExperience={[MockFunction]}
    employer="Oracle"
    fromMonth="sep"
    fromYear="2017"
    jobTitle="Software Engineer"
    location="Cambridge, MA"
    toMonth="sep"
    toYear="2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="selected-education"
        >
          <WithStyles(IconButton)
            aria-label="Delete"
            className="nabi-float-right"
            onClick={[Function]}
          >
            <pure(Delete) />
          </WithStyles(IconButton)>
          <WithStyles(Typography)
            className="nabi-margin-top-small"
            variant="body2"
          >
            SOFTWARE ENGINEER
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className=""
          >
            Oracle
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className=""
          >
            Sep, 2017 - Sep, 2018
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className=""
          >
            Cambridge, MA
          </WithStyles(Typography)>
        </div>,
        <WithStyles(Divider)
          className="nabi-margin-top-small"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>,
            <WithStyles(Typography)
              className="nabi-margin-top-small"
              variant="body2"
            >
              SOFTWARE ENGINEER
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className=""
            >
              Oracle
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className=""
            >
              Sep, 2017 - Sep, 2018
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className=""
            >
              Cambridge, MA
            </WithStyles(Typography)>,
          ],
          "className": "selected-education",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <pure(Delete) />,
              "className": "nabi-float-right",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "SOFTWARE ENGINEER",
              "className": "nabi-margin-top-small",
              "variant": "body2",
            },
            "ref": null,
            "rendered": "SOFTWARE ENGINEER",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Oracle",
              "className": "",
            },
            "ref": null,
            "rendered": "Oracle",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sep, 2017 - Sep, 2018",
              "className": "",
            },
            "ref": null,
            "rendered": "Sep, 2017 - Sep, 2018",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cambridge, MA",
              "className": "",
            },
            "ref": null,
            "rendered": "Cambridge, MA",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "nabi-margin-top-small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="selected-education"
          >
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>
            <WithStyles(Typography)
              className="nabi-margin-top-small"
              variant="body2"
            >
              SOFTWARE ENGINEER
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className=""
            >
              Oracle
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className=""
            >
              Sep, 2017 - Sep, 2018
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className=""
            >
              Cambridge, MA
            </WithStyles(Typography)>
          </div>,
          <WithStyles(Divider)
            className="nabi-margin-top-small"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Delete"
                className="nabi-float-right"
                onClick={[Function]}
              >
                <pure(Delete) />
              </WithStyles(IconButton)>,
              <WithStyles(Typography)
                className="nabi-margin-top-small"
                variant="body2"
              >
                SOFTWARE ENGINEER
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                className=""
              >
                Oracle
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                className=""
              >
                Sep, 2017 - Sep, 2018
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                className=""
              >
                Cambridge, MA
              </WithStyles(Typography)>,
            ],
            "className": "selected-education",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <pure(Delete) />,
                "className": "nabi-float-right",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "SOFTWARE ENGINEER",
                "className": "nabi-margin-top-small",
                "variant": "body2",
              },
              "ref": null,
              "rendered": "SOFTWARE ENGINEER",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Oracle",
                "className": "",
              },
              "ref": null,
              "rendered": "Oracle",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sep, 2017 - Sep, 2018",
                "className": "",
              },
              "ref": null,
              "rendered": "Sep, 2017 - Sep, 2018",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cambridge, MA",
                "className": "",
              },
              "ref": null,
              "rendered": "Cambridge, MA",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "nabi-margin-top-small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
